blockchain/txvalidator.go

```diff
@@ -163,14 +163,15 @@ func CheckTransactionInput(txn *Transaction) error {
 	if len(txn.Inputs) <= 0 {
 		return errors.New("transaction has no inputs")
 	}
-	for i, utxoin := range txn.Inputs {
-		if utxoin.Previous.TxID.IsEqual(EmptyHash) && (utxoin.Previous.Index == math.MaxUint16) {
+	inputSet := make(map[string]bool)
```

Comment:
Normally we use a map with empty struct as value to du duplicate element check. This is not a rule in go programing language but a common usage.

Reply:
fixed in the next commit.